File: USACO/teleportation.java
import java.io.*;
import java.util.StringTokenizer;

//old java code ignore
public class teleportation {
    public static void main(String[] args) throws IOException
    {
        BufferedReader r = new BufferedReader(new FileReader("teleport.in"));
        PrintWriter pw = new PrintWriter("teleport.out");
        StringTokenizer st = new StringTokenizer(r.readLine());
        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());
        int x = Integer.parseInt(st.nextToken());
        int y = Integer.parseInt(st.nextToken());

        int ans = Math.abs(a - b);
        ans = Math.min(ans, Math.abs(a - x) + Math.abs(b - y));
        ans = Math.min(ans, Math.abs(a - y) + Math.abs(b - x));
        pw.println(ans);
        pw.close();
        r.close();
    }
}

File: USACO/teleportation.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    cin.tie(0)->sync_with_stdio(0);
    freopen("teleport.in", "r", stdin);
    freopen("teleport.out", "w", stdout);

    int a, b, x, y;
    cin >> a >> b >> x >> y;

    int ans = abs(a - b);
    ans = min(ans, abs(a - x) + abs(b - y));
    ans = min(ans, abs(a - y) + abs(b - x));
    cout << ans << "\n";
}